relentless/optimize/method.py

```diff
@@ -121,11 +131,144 @@ def abs_tol(self, value):
         else:
             self._abs_tol = abs_tol
 
+class LineSearch:
+    r"""Line search algorithm.
+
+    For an :class:`~relentless.optimize.ObjectiveFunction` :math:`f\left(\mathbf{x}\right)`,
+    the line search algorithm seeks to take a single, optimally-sized step towards
+    the minimum of the function. (Does not necessarily converge to minimum.)
+
+    A search interval :math:`\mathbf{d}` is specified, and normalized to :math:`\hat{\mathbf{d}}`:
+
+    .. math::
+
+        \mathbf{d} = \mathbf{x}_{end} - \mathbf{x}_{start}
+
+        \hat{\mathbf{d}} = \frac{\mathbf{d}}{\lVert\mathbf{d}\rVert}
+
+    Then, a 'target' value :math:`t` is defined as:
+
+    .. math::
+
+        t = -\hat{\mathbf{d}} \cdot \nabla{f\left(\mathbf{x}\right)}
+
+    The target at the start of the search interval is always negative. If the
```

Comment:
```suggestion
    Because :math:`\hat{\mathbf{d}}`: is a descent direction, the target at the start of the search interval is always positive. If the
```